### PR TITLE
[RESTEASY-1974] Checkstyle - final fixes

### PR DESCRIPTION
**Pull request by tterem**
Fixes according to rules in checkstyle.xml, mostly:
* trailing spaces
* modifier order
* final arguments

Update checkstyle.xml
* allow System.out/err, printStackTrace in comments
* allow to suppress checkstyle using 'CHECKSTYLE.OFF' and 'CHECKSTYLE.ON'

Update pom.xml
* run checkstyle check by default

Jira: https://issues.jboss.org/browse/RESTEASY-1974
3.6 PR: https://github.com/resteasy/Resteasy/pull/1768

### CODE REVIEW

File: resteasy-wadl/src/main/java/org/jboss/resteasy/wadl/ResteasyWadlWriter.java [hunk 1]
```diff
@@ -395,7 +395,9 @@ public Result createOutput(final String namespaceUri, final String suggestedFile
                    Iterator<Include> iter = grammars.getInclude().iterator();
                    while (iter.hasNext()) {
                        for (Doc doc : iter.next().getDoc()) {
-                           if ("Generated".equals(doc.getTitle())) ;
+                           // CHECKSTYLE.OFF: EmptyStatement
+                           if ("Generated".equals(doc.getTitle()));
```
Comment (tterem) [inline resteasy-wadl/src/main/java/org/jboss/resteasy/wadl/ResteasyWadlWriter.java]: Checkstyle reports "empty statement" here. I wasn't sure what to do with it.
I mentioned it here: https://github.com/resteasy/Resteasy/pull/1649#discussion_r229681375

File: resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java [hunk 2]
```diff
@@ -128,24 +125,28 @@ private SseEventSourceImpl(final WebTarget target, String name, long reconnectDe
...
          ScheduledExecutorService scheduledExecutor = null;
          if (target instanceof ResteasyWebTarget)
          {
             scheduledExecutor = ((ResteasyWebTarget) target).getResteasyClient().getScheduledExecutor();
          }
-         this.sseEventSourceScheduler = new SseEventSourceScheduler(scheduledExecutor, name);
+         if (name != null) {
+            this.sseEventSourceScheduler = new SseEventSourceScheduler(executor, name);
+         } else {
+            this.sseEventSourceScheduler = new SseEventSourceScheduler(executor, String.format("sse-event-source(%s)", target.getUri()));
```
[the remark this answers is not in the document]
Comment (tterem) [reply resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java]: You are right, I fixed it. Thanks for review.

**Comment by tterem** [comment]
I fixed that one error and made rebase.

**Comment by asoldano** [comment]
@tterem , is the last commit (https://github.com/resteasy/Resteasy/pull/1763/commits/3367e58cb0cb0a677dc07af67b3ba67de7b55739) the only addition/change after my previous review? Are you done with the changes ? (IOW, can I merge now ?)

**Comment by tterem** [comment]
@asoldano I had to do rebase again and fix checkstyle errors introduced in yesterday commits. These fixes are in last(https://github.com/resteasy/Resteasy/pull/1763/commits/b8f59b33de2ac6262d5d4d525e6374e6d665a14d) commit. It was mostly trailing space errors and redundant public modifiers in interfaces. You can merge this.

**Pull request merged**